x-pack/plugins/monitoring/public/components/metricbeat_migration/flyout/flyout.js

```diff
@@ -0,0 +1,345 @@
+    if (removeInterval) {
+      clearInterval(this.checkInterval);
+      this.clearInterval = null;
+    }
+
+    if (setupInterval) {
+      this.checkInterval = setInterval(async () => {
+        await this.checkForMigrationStatus();
+      }, AUTO_CHECK_INTERVAL_IN_MS);
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.checkInterval);
+  }
+
+  checkForMigrationStatus = async () => {
+    this.setState({ checkingMigrationStatus: true });
+    await ensureMinimumTime(this.props.updateProduct(), 1000);
+    this.setState(state => ({
+      ...state,
+      checkingMigrationStatus: false,
+      checkedStatusByStep: {
+        ...state.checkedStatusByStep,
+        [this.state.activeStep]: true,
+      }
+    }));
+  }
+
+  setEsMonitoringUrl = esMonitoringUrl => {
+    storage.set(ES_MONITORING_URL_KEY, esMonitoringUrl);
+    this.setState({ esMonitoringUrl });
+  }
+
+  renderActiveStep() {
+    const { product, productName, onClose, meta, instance } = this.props;
+    const {
+      activeStep,
+      esMonitoringUrl,
+      checkedStatusByStep,
+      checkingMigrationStatus,
+    } = this.state;
+
+    switch (activeStep) {
+      case INSTRUCTION_STEP_SET_MONITORING_URL:
+        return (
+          <EuiForm>
+            <EuiFormRow
+              fullWidth
+              label={i18n.translate('xpack.monitoring.metricbeatMigration.flyout.step1.monitoringUrlLabel', {
+                defaultMessage: 'Monitoring cluster URL'
+              })}
+              helpText={i18n.translate('xpack.monitoring.metricbeatMigration.flyout.step1.monitoringUrlHelpText', {
+                defaultMessage: `This is typically a single instance, but if you have multiple, enter all of instance urls comma-separated.
+                Keep in mind that the running metricbeat instance will need to be able to communicate with these Elasticsearch servers.`
+              })}
+            >
+              <EuiFieldText
+                value={esMonitoringUrl}
+                onChange={e => this.setEsMonitoringUrl(e.target.value)}
+              />
+            </EuiFormRow>
+          </EuiForm>
+        );
+      case INSTRUCTION_STEP_ENABLE_METRICBEAT:
+      case INSTRUCTION_STEP_DISABLE_INTERNAL:
+        const instructionSteps = getInstructionSteps(productName, product, activeStep, meta, {
+          doneWithMigration: onClose,
+          esMonitoringUrl,
+          checkForMigrationStatus: this.checkForMigrationStatus,
+          checkingMigrationStatus,
+          hasCheckedStatus: checkedStatusByStep[activeStep],
+          autoCheckIntervalInMs: AUTO_CHECK_INTERVAL_IN_MS,
+        });
+
+        let calloutTitle = null;
+        if (productName === KIBANA_SYSTEM_ID) {
+          calloutTitle = (
+            <FormattedMessage
+              id="xpack.monitoring.metricbeatMigration.flyout.kibana.migrationDescriptionLabel"
+              defaultMessage="Migrating instance {instance}"
```

Comment:
Can you just move the instance name to the flyout title, I completely missed this thinking it was just an informational bubble, not that it was telling me which one I've selected to migrate.

<img src="https://d.pr/free/i/1SoZgB+" width="50%" />

Reply:
<img width="964" alt="Screen Shot 2019-06-14 at 9 59 14 AM" src="https://user-images.githubusercontent.com/56682/59514582-2989b600-8e8b-11e9-80d0-2fda9f0cd3a5.png">